_posts/2024-08-02-gsoc2024_talagayev.md

```diff
@@ -0,0 +1,193 @@
+features in the `IpyWidgets` visualization, which would give the user important information in regards to the characteristics of the molecule that is being visualized.
+
+## What I did
+
+I would divide this section into separate parts, each highlighting one of the important aspects of the work.
+
+### Main Visualization Code
+
+The main part of the code consisted of setting up the code for the visualization, which would allow the display of the molecule in a jupyter notebook. This is performed in the 
+`mdonatello.py`, which is mainly responsible for the visualization with it having the `class MoleculeVisualizer` that is the main class and responsible for the visualization with
+the functions in this class initializing the `IpyWidgets` interactive widgets such as a bar, that allows the user to select the molecule that he wants to display and also the
+interactive checkboxes with each having a feature, that I would explain in a separate section.
+
+This main code was the first part of the code that I designed and was in the beginning only able to display the molecule and had only one interactive checkbox, which displayed the atom
+indices of the molecules, but as more and more features were added the visualization grew bigger and had more features that it needed to display.
+
+### Feature Addition
+
+The feature addition was the next step that I was working on and was mainly the biggest part of the project since there was always an additional interesting feature that could be added
+that was proposed either by myself or one of the mentors.
+
+The first feature addition that I was working on were the physiochemical properties as well as the number of hydrogen bond donors and acceptors of the molecule.
+
+After it was done I started adding features, which would by selecting the checkbox highlight the features mentioned in the checkbox. For example, I implemented pharmacophore checkboxes,
+which by selection would highlight the selected pharmacophores, thus by selecting a hydrophobic feature, the atoms of the molecule that are hydrophobic via the `RDKit` pharmacophore
+recognition were highlighted in the assigned colors.
+
+Here is a short summary of features that I added for the visualization, which would show a value for the corresponding feature or adjust the figure of the molecule to display the values:
+
+- **Phyisochemical Properties**, **Atom Indices**, **Partial Charges**, **Hydrogen Bond Donors&Acceptors**
+
+And here is a short summary of features, that will highlight a specific atoms and bonds corresponding to their features if they are selected:
+
+- **Rotatable Bonds**, **Partial Charge Heatmap**, **Functional Groups**, **Stereocenters**, **Murcko Scaffold**, **Pharmacophores**
+
+Here I would also address that for the pharmacophores there is a separate checkbox for each of the possible pharmacophore features, thus you can select the specific pharmaocphore feature
```

Comment:
```suggestion
There is a separate checkbox for each pharmacophoric feature, thus you can select the specific feature
```
And you can put that sentence directly with the `- **Pharmacophores**` bullet point